template/upstream.c.erb

```diff
@@ -27,22 +27,29 @@ static void btx_dispatch_<%= e.name_sanitized %>(
   const bt_clock_snapshot *clock_snapshot = bt_message_event_borrow_default_clock_snapshot_const(upstream_message);
   bt_clock_snapshot_get_ns_from_origin(clock_snapshot, &_timestamp);
   <% end %>
-  // Call all the callbacks who where registered
-  // Their type are declared in 'upstream.h'
-  <%= e.name_sanitized %>_callback_f **p = NULL;
-  while ((p = utarray_next(callbacks->event_callbacks, p))) {
-    (*p)((void *)common_data,
-         common_data
-             ->usr_data<%= ', _timestamp' if e.default_clock_class %><%= e.args.map{ |s| s.name }.join_with_prefix(", ") %>);
+
+  /* Call registered matching callbacks */
+  <% e.dispatchers.each do |d| %>
```

Comment:
Let's sort the dispatchers so we give some control in user land

Reply:
The event dispacther is always the first one. Matching dispacthers will be called in the same order as defined in the YAML, are you OK with this in place of sorting?